cloudigrade/account/views.py

```diff
@@ -73,11 +73,17 @@ def get_queryset(self):
         """Get the queryset filtered to appropriate user."""
         user = self.request.user
         if not user.is_superuser:
-            return self.queryset.filter(instance__account__user__id=user.id)
+            self.queryset = self.queryset.filter(
```

Comment:
This looks like it's generally on the right track, but we should assign the filtered queryset to a local variable (perhaps just `queryset`) instead of overriding `self.queryset` and return that local variable at the end. By clobbering the instance variable as is currently here, later calls to `get_queryset` will already have these filters already applied at the start which may be wrong depending on their context.

Reply:
@infinitewarp Fixed.